### ISSUE TITLE
Test suite leaks file descriptors.

### ISSUE THREAD

**Issue by mdickinson** [enthought/envisage#186]
The IPython-related tests leak approximately 20 file descriptors per test. Once there are enough tests, this causes the OS to run out of file descriptors. We're seeing this particularly on Travis CI, where the number of available file descriptors per process is apparently relatively small.

**Comment by mdickinson** [comment]
Here's the output from running with `haas` and the haas plugins from `enthought/haas-plugins`:
```
(envisage-test-3.6-null)bash-3.2$ haas --track-fds envisage >/dev/null
Number of open file descriptors at start of test run: 10
.test_ctrl_c_message_suppressed (envisage.plugins.ipython_kernel.tests.test_internal_ipkernel.TestInternalIPKernel) changed number of open file descriptors from 10 to 47
.test_initial_namespace (envisage.plugins.ipython_kernel.tests.test_internal_ipkernel.TestInternalIPKernel) changed number of open file descriptors from 47 to 75
.test_io_pub_thread_stopped (envisage.plugins.ipython_kernel.tests.test_internal_ipkernel.TestInternalIPKernel) changed number of open file descriptors from 75 to 93
.test_lifecycle (envisage.plugins.ipython_kernel.tests.test_internal_ipkernel.TestInternalIPKernel) changed number of open file descriptors from 93 to 119
.test_shutdown_closes_console_pipes (envisage.plugins.ipython_kernel.tests.test_internal_ipkernel.TestInternalIPKernel) changed number of open file descriptors from 118 to 141
.test_shutdown_restores_output_streams (envisage.plugins.ipython_kernel.tests.test_internal_ipkernel.TestInternalIPKernel) changed number of open file descriptors from 139 to 167
.test_import_from_api (envisage.plugins.ipython_kernel.tests.test_ipython_kernel_plugin.TestIPythonKernelPlugin) changed number of open file descriptors from 167 to 163
.test_kernel_namespace_extension_point (envisage.plugins.ipython_kernel.tests.test_ipython_kernel_plugin.TestIPythonKernelPlugin) changed number of open file descriptors from 162 to 185
.test_kernel_service (envisage.plugins.ipython_kernel.tests.test_ipython_kernel_plugin.TestIPythonKernelPlugin) changed number of open file descriptors from 185 to 209
Number of open file descriptors at end of test run: 208

----------------------------------------------------------------------
Ran 9 tests in 0.208s

OK
```

**Comment by mdickinson** [comment]
Closed by #188
